cli/version/version_match.go

```diff
@@ -0,0 +1,174 @@
+package version
+
+import (
+	"fmt"
+
+	"github.com/tarantool/tt/cli/util"
+)
+
+type requiredField uint8
+type requiredFields []requiredField
+
+const (
+	// Values according order of Fields in regex from `matchVersionParts()`.
+	requiredFieldBuildName requiredField = iota
+	requiredFieldMajor
+	requiredFieldMinor
+	requiredFieldPatch
+	requiredFieldReleaseType
+	requiredFieldReleaseNum
+	requiredFieldAdditional
+	requiredFieldHash
+	requiredFieldRevision
+
+	countOfRequiredFields
+)
+
+// NotFoundError is sentinel error for case when required version not found.
+type NotFoundError struct {
+	expected string
+}
+
+func (e NotFoundError) Error() string {
+	return fmt.Sprintf("not matched with: %s", e.expected)
+}
+
+// exploreMatchVersion examines the given string and retrieves the version components,
+// that define the rules for checking for version consistency.
+func exploreMatchVersion(verStr string) (Version, requiredFields, error) {
+	var version Version
+	fields := make(requiredFields, 0, countOfRequiredFields)
+
+	matches, err := matchVersionParts(verStr, false)
+	if err != nil {
+		return version, fields, err
+	}
+
+	version.BuildName = matches["buildname"]
+	if version.BuildName != "" {
+		fields = append(fields, requiredFieldBuildName)
+	}
+
+	if matches["major"] != "" {
+		if version.Major, err = util.AtoiUint64(matches["major"]); err != nil {
+			return version, fields, fmt.Errorf("can't parse Major: %w", err)
+		}
+		fields = append(fields, requiredFieldMajor)
+	}
+
+	if matches["minor"] != "" {
+		if matches["major"] == "" {
+			return version, fields, fmt.Errorf("parse Minor requires Major: %w", err)
+		}
+		if version.Minor, err = util.AtoiUint64(matches["minor"]); err != nil {
+			return version, fields, fmt.Errorf("can't parse Minor: %w", err)
+		}
+		fields = append(fields, requiredFieldMinor)
+	}
+
+	if matches["patch"] != "" {
+		if matches["minor"] == "" {
+			return version, fields, fmt.Errorf("parse Patch requires Minor: %w", err)
+		}
```

Comment:
Due to the way regex works, this case is not possible. This branch of checks in the code is currently unreachable. It is added only in case regex will changes.

Reply:
I think we need a prettier error for the user here. Something like:

```minor version requires major to be specified```

The error looks confusing from a user side. There are too many implementation details that the user will not understand.

Reply:
I catch that this case is not possible. I mean that the error a bit confusing. We type something like `.<patch>` and the output says `parse Minor requires Major`.

Reply:
> when we type something like `.<patch>`

We type `.<digit>` and what we put into it nobody knows. In our understanding, it can be either a `.patch` or a `.minor`. But the current regexp considers such input as a `.minor`.

I hope that Oleg's suggestion to change the message will help solve the problem.

Reply:
Ok, let it be.